### PR TITLE
CORE-2932: request synchronisation

### PR DESCRIPTION
**Pull request by yift-r3**
Associated runtime PR in https://github.com/corda/corda-runtime-os/pull/1960

Makes the `hashCheck` optional.

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/p2p/SignedMemberships.avsc [hunk 1]
```diff
@@ -7,7 +7,7 @@
     {
       "name": "hashCheck",
       "doc": "Hash value to help identify out-of-sync members.",
-      "type": "net.corda.data.crypto.SecureHash"
+      "type": ["null", "net.corda.data.crypto.SecureHash"]
```
Comment (nikinagy) [inline data/avro-schema/src/main/resources/avro/net/corda/data/membership/p2p/SignedMemberships.avsc:10]: just a question: why was this change needed?

Comment (yift-r3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/membership/p2p/SignedMemberships.avsc:10]: From the [ticket](https://r3-cev.atlassian.net/browse/CORE-2932):

>  ...start sync when no hash from the MGM is available

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — APPROVED** [review]
